.eslintrc.json

```diff
@@ -1,32 +1,47 @@
 {
+	"root": true, // this is root configuration for project
+	"parser": "@typescript-eslint/parser",
+	"parserOptions": {
+		"ecmaVersion": 2022,
+		"sourceType": "commonjs",
+		"project": "./tsconfig.json",
+		"ecmaFeatures": {
+			"impliedStrict": true
+		}
+	},
+
 	"env": {
 		"browser": true,
-		"es2020": true,
+		"es2022": true,
 		"mocha": true
 	},
 	"extends": [
 		"eslint:recommended",
 		"plugin:@typescript-eslint/recommended",
 		"plugin:import/typescript"
 	],
-	"parser": "@typescript-eslint/parser",
-	"parserOptions": {
-		"ecmaVersion": 2020,
-		"sourceType": "script",
-		"ecmaFeatures": {
-			"impliedStrict": true
-		},
-		"project": "./tsconfig.json"
-	},
+	
 	"plugins": [
 		"@typescript-eslint",
-		"eslint-plugin-import"
+		"eslint-plugin-import" //  required by no-extraneous-dependencies
 	],
 	"rules": {
 		"@typescript-eslint/no-var-requires": "off", /** allows require statements outside of imports **/
 		"no-async-promise-executor": "off", /* Deactivated for now as i do not know how to fix it safely*/
 		"@typescript-eslint/no-namespace": "off",
 		"@typescript-eslint/no-explicit-any": "off",
+		"@typescript-eslint/naming-convention": [ 
+			"error",
+			{
+			  "selector": "variable",
+			  "format": ["camelCase", "UPPER_CASE"]
+			}
+		  ],
+		"@typescript-eslint/semi": "warn",
+		"curly": "warn",
+		"eqeqeq": "warn",
+		"@typescript-eslint/no-unused-vars": "warn",
+		"no-throw-literal": "warn",
```

Comment:
All of them are required by the "recommended" config and we set to warn to not fail the build currently?

Reply:
we want them as warning mainly for local development purpose - it's really not necessary to get errors because of unused variable or missing semicolon, on the other hand it's necessary to watch these rules so we won't accept PR with errors